// @rebelfi/sdk/dist/src/api/allocations.js
import { unwrapResponse } from '../client.js';
export class AllocationsAPI {
    constructor(client) {
        this.client = client;
    }
    /**
     * List all allocations for a wallet.
     * Accepts AllocationListRequest (POST, supports walletId/userId) or
     * legacy AllocationQuery (GET, walletAddress only).
     */
    async list(query) {
        // If it has walletId or userId, use the new POST endpoint
        if ('walletId' in query || 'userId' in query) {
            const response = await this.client.post('/v1/allocations', query);
            return unwrapResponse(response);
        }
        // Legacy: use POST with walletAddress
        const response = await this.client.post('/v1/allocations', query);
        return unwrapResponse(response);
    }
    /**
     * Get allocation by strategy ID
     * @param strategyId Strategy ID
     * @param wallet Wallet address or wallet ID
     */
    async get(strategyId, wallet) {
        const params = typeof wallet === 'string'
            ? { walletAddress: wallet }
            : { walletId: wallet };
        const response = await this.client.get(`/v1/allocations/strategy/${strategyId}`, { params });
        return unwrapResponse(response);
    }
    /**
     * @deprecated Use get(strategyId, wallet) instead
     */
    async getByVenue(venueId, walletAddress) {
        const response = await this.client.get(`/v1/allocations/${venueId}`, {
            params: { walletAddress }
        });
        return unwrapResponse(response);
    }
    /**
     * Get earnings history for a wallet with daily granularity
     */
    async earnings(query) {
        const response = await this.client.get('/v1/allocations/earnings', { params: query });
        return unwrapResponse(response);
    }
}
