// @rebelfi/sdk/dist/src/types.js
// ============================================
// CONFIG
// ============================================
// ============================================
// COMMON TYPES
// ============================================
export var Blockchain;
(function (Blockchain) {
    Blockchain["SOLANA"] = "solana";
    Blockchain["POLYGON"] = "polygon";
    Blockchain["ETHEREUM"] = "ethereum";
})(Blockchain || (Blockchain = {}));
export var VenueStatus;
(function (VenueStatus) {
    VenueStatus["ACTIVE"] = "ACTIVE";
    VenueStatus["PAUSED"] = "PAUSED";
    VenueStatus["DEPRECATED"] = "DEPRECATED";
})(VenueStatus || (VenueStatus = {}));
export var StrategyStatus;
(function (StrategyStatus) {
    StrategyStatus["ACTIVE"] = "ACTIVE";
    StrategyStatus["PAUSED"] = "PAUSED";
})(StrategyStatus || (StrategyStatus = {}));
export var TransactionStatus;
(function (TransactionStatus) {
    TransactionStatus["UNSIGNED"] = "UNSIGNED";
    TransactionStatus["PENDING"] = "PENDING";
    TransactionStatus["CONFIRMED"] = "CONFIRMED";
    TransactionStatus["FAILED"] = "FAILED";
})(TransactionStatus || (TransactionStatus = {}));
export var OperationType;
(function (OperationType) {
    OperationType["SUPPLY"] = "SUPPLY";
    OperationType["UNWIND"] = "UNWIND";
})(OperationType || (OperationType = {}));
export var OperationStatus;
(function (OperationStatus) {
    OperationStatus["PENDING"] = "PENDING";
    OperationStatus["AWAITING_SIGNATURE"] = "AWAITING_SIGNATURE";
    OperationStatus["SUBMITTED"] = "SUBMITTED";
    OperationStatus["CONFIRMED"] = "CONFIRMED";
    OperationStatus["FAILED"] = "FAILED";
    OperationStatus["CANCELLED"] = "CANCELLED";
})(OperationStatus || (OperationStatus = {}));
// ============================================
// ERROR TYPES
// ============================================
/**
 * SDK error codes for machine-readable error handling.
 * These mirror the backend SdkErrorCode enum.
 */
export var ErrorCode;
(function (ErrorCode) {
    // Validation errors
    ErrorCode["INVALID_AMOUNT"] = "INVALID_AMOUNT";
    ErrorCode["INVALID_ADDRESS"] = "INVALID_ADDRESS";
    ErrorCode["INVALID_TOKEN"] = "INVALID_TOKEN";
    // Business logic errors
    ErrorCode["INSUFFICIENT_BALANCE"] = "INSUFFICIENT_BALANCE";
    ErrorCode["STRATEGY_NOT_ACTIVE"] = "STRATEGY_NOT_ACTIVE";
    ErrorCode["ALLOCATION_NOT_FOUND"] = "ALLOCATION_NOT_FOUND";
    ErrorCode["OPERATION_EXPIRED"] = "OPERATION_EXPIRED";
    ErrorCode["OPERATION_ALREADY_SUBMITTED"] = "OPERATION_ALREADY_SUBMITTED";
    ErrorCode["TOKEN_MISMATCH"] = "TOKEN_MISMATCH";
    ErrorCode["INVALID_OPERATION_STATUS"] = "INVALID_OPERATION_STATUS";
    ErrorCode["OPERATION_IN_PROGRESS"] = "OPERATION_IN_PROGRESS";
    // Resource errors
    ErrorCode["VENUE_NOT_FOUND"] = "VENUE_NOT_FOUND";
    ErrorCode["STRATEGY_NOT_FOUND"] = "STRATEGY_NOT_FOUND";
    ErrorCode["OPERATION_NOT_FOUND"] = "OPERATION_NOT_FOUND";
    ErrorCode["TRANSACTION_NOT_FOUND"] = "TRANSACTION_NOT_FOUND";
    ErrorCode["WALLET_NOT_FOUND"] = "WALLET_NOT_FOUND";
    ErrorCode["ORGANIZATION_NOT_FOUND"] = "ORGANIZATION_NOT_FOUND";
    ErrorCode["TOKEN_NOT_FOUND"] = "TOKEN_NOT_FOUND";
    // Simulation errors
    ErrorCode["INSUFFICIENT_GAS"] = "INSUFFICIENT_GAS";
    ErrorCode["SIMULATION_FAILED"] = "SIMULATION_FAILED";
    // Client-side errors
    ErrorCode["INVALID_API_KEY"] = "INVALID_API_KEY";
    ErrorCode["API_KEY_DISABLED"] = "API_KEY_DISABLED";
    ErrorCode["RATE_LIMIT_EXCEEDED"] = "RATE_LIMIT_EXCEEDED";
    ErrorCode["TIMEOUT"] = "TIMEOUT";
    ErrorCode["NETWORK_ERROR"] = "NETWORK_ERROR";
    ErrorCode["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
})(ErrorCode || (ErrorCode = {}));
/**
 * Transaction failure codes for blockchain-specific errors.
 */
export var TransactionFailureCode;
(function (TransactionFailureCode) {
    /** Smart contract execution reverted */
    TransactionFailureCode["REVERTED"] = "REVERTED";
    /** Insufficient gas */
    TransactionFailureCode["OUT_OF_GAS"] = "OUT_OF_GAS";
    /** Not included in block within timeout */
    TransactionFailureCode["TIMEOUT"] = "TIMEOUT";
    /** Nonce already used */
    TransactionFailureCode["NONCE_TOO_LOW"] = "NONCE_TOO_LOW";
    /** Not enough native token for gas */
    TransactionFailureCode["INSUFFICIENT_FUNDS"] = "INSUFFICIENT_FUNDS";
    /** Transaction replaced by another */
    TransactionFailureCode["REPLACED"] = "REPLACED";
    /** Dropped from mempool */
    TransactionFailureCode["DROPPED"] = "DROPPED";
    /** Unknown failure */
    TransactionFailureCode["UNKNOWN"] = "UNKNOWN";
})(TransactionFailureCode || (TransactionFailureCode = {}));
export class RebelfiError extends Error {
    constructor(message, statusCode, code, details) {
        super(message);
        this.statusCode = statusCode;
        this.code = code;
        this.details = details;
        this.name = 'RebelfiError';
    }
    /**
     * Check if error matches a specific error code
     */
    is(code) {
        return this.code === code;
    }
}
